### PR TITLE
Issue #3000534 by chr.fritsch: Add a facet select2 widget

### PR DESCRIPTION
**Pull request by chrfritsch**

### CODE REVIEW

**Comment by chrfritsch** [comment]
@dbosen I need to download search_api and facets on Travis for testing the new sub-module. Is that somehow possible with thunder/travis?

File: modules/select2_facets/src/Plugin/facets/widget/Select2Widget.php [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+    foreach ($facet->getResults() as $result) {
+      if (empty($result->getUrl())) {
+        continue;
+      }
+      // When the facet is being build in an AJAX request, and the facetsource
+      // is a block, we need to update the url to use the current request url.
+      if ($result->getUrl()->isRouted() && $result->getUrl()->getRouteName() === 'facets.block.ajax') {
+        $request = \Drupal::request();
+        $url_object = \Drupal::service('path.validator')->getUrlIfValid($request->getPathInfo());
```
Comment (dbosen) [inline modules/select2_facets/src/Plugin/facets/widget/Select2Widget.php]: Please inject services

File: modules/select2_facets/src/Plugin/facets/widget/Select2Widget.php [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    $items = [];
+    $active_items = [];
+
+    foreach ($facet->getResults() as $result) {
+      if (empty($result->getUrl())) {
+        continue;
+      }
+      // When the facet is being build in an AJAX request, and the facetsource
+      // is a block, we need to update the url to use the current request url.
+      if ($result->getUrl()->isRouted() && $result->getUrl()->getRouteName() === 'facets.block.ajax') {
```
Comment (dbosen) [inline modules/select2_facets/src/Plugin/facets/widget/Select2Widget.php]: How can I get into this?

File: modules/select2_facets/src/Plugin/facets/widget/Select2Widget.php [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+      'request' => serialize(\Drupal::request()),
+      'match_operator' => $this->getConfiguration()['match_operator'],
+    ];
+
+    // Store the selection settings in the key/value store and pass a hashed key
+    // in the route parameters.
+    $data = serialize($selection_settings) . $this->facet->getFacetSourceId() . $this->facet->id();
+    $selection_settings_key = Crypt::hmacBase64($data, Settings::getHashSalt());
+
+    $key_value_storage = \Drupal::keyValue('entity_autocomplete');
```
Comment (dbosen) [inline modules/select2_facets/src/Plugin/facets/widget/Select2Widget.php]: Another not injected service

**Review by dbosen — CHANGES_REQUESTED** [review]

**Review by dbosen — APPROVED** [review]

**Pull request merged**